### PR TITLE
Fix torrent health conversion

### PR DESCRIPTION
**Pull request by ichorid**
This adds conversion of the health info from 7.2.2 database to Pony. Also, it fixes a bug with torrents conversion introduced in #4572 that resulted in skipping conversion of most entries.

### CODE REVIEW

**Comment by ichorid** [comment]
retest this please

File: Tribler/Core/Upgrade/db72_to_pony.py [hunk 1]
```diff
@@ -292,23 +302,22 @@ def convert_async(self, add_to_pony, get_old_stuff, total_to_convert, offset=0,
...
-            except (TransactionIntegrityError, CacheIndexError):
-                pass
+            with db_session:
+                for (torrent_dict, health) in batch:
+                    try:
+                        torrent = add_to_pony(torrent_dict)
+                        if torrent and health:
+                            torrent.health.set(**health)
+                    except:
+                        pass
```
Comment (qstokkink) [inline Tribler/Core/Upgrade/db72_to_pony.py]: Could you hook in a logging message here for easier debugging in the future?

File: Tribler/Core/Upgrade/db72_to_pony.py [hunk 2]
```diff
@@ -172,49 +174,58 @@ def get_old_torrents(self, personal_channel_only=False, batch_size=10000, offset
...
+                leechers = int(num_leechers or 0)
+                last_tracker_check = int(last_tracker_check or 0)
                 health_dict = {
-                    "seeders": int(num_seeders or 0),
-                    "leechers": int(num_leechers or 0),
-                    "last_check": int(last_tracker_check or 0)}
+                    "seeders": seeders,
+                    "leechers": leechers,
+                    "last_check": last_tracker_check
+                } if (last_tracker_check >= 0 and seeders >= 0 and leechers >= 0) else None
```
Comment (xoriole) [inline Tribler/Core/Upgrade/db72_to_pony.py]: Putting `and` might not be the correct condition here

Comment (ichorid) [reply Tribler/Core/Upgrade/db72_to_pony.py]: Why so? The reason for this check is to only convert health entries that have correct (non-negative) attribute values.

Comment (xoriole) [reply Tribler/Core/Upgrade/db72_to_pony.py]: You're right.

**Pull request merged**